### DISCUSSION TITLE
Where Did You Hide Current Temperature for a lattitude and longitude using JavaScript?

### DISCUSSION THREAD

**Discussion by jkreski** [weather-gov/api#723] [Q&A] [answered]
I have been looking through the API and found how to fetch what looks to be forecast data for a gridpoint.  It returns something like 100  temperatures, none of which are current.  Great data but I am looking for the current conditions for a location.  I am using JavaScript and am relatively new to the language but have taken a bootcamp course in it and have other programming experiences.  Thanks!
https://api.weather.gov/gridpoints/APX/14,39 is a path I have tried for example...

**Comment by JJKraw** [comment, marked as answer]
The endpoint you have specified is a _forecast_ endpoint. It doesn't contain current conditions. Also, there will not be such data available for a random gridpoint because for most there isn't a reporting station that could provide the data.

Current conditions for locations with reporting stations are available under stations/{stationId}/observations/latest

**Reply by jkreski** [reply]
Thank you!  I can find a temperature for Green Bay, WI with this link:
https://api.weather.gov/stations/KGRB/observations/latest/ but if I use this link to find the nearest office to my location over here in Michigan: https://api.weather.gov/points/44.630978,-86.072480 it gives me a CWA of APX which I assume is the office ID I need to use in the first link.  If I use either that or KAPX it throws an error of not found.  Not sure CWA is the office ID.  How can I navigate from my first link to the office ID so I can fetch my current temperature?  Thanks!  We're making progress...

**Reply by dwhitemv25** [reply]
If you're looking at the `/points` output, follow the `observationStations` URL to get the list of stations sorted by distance from the point. Take the `stationIdentifier` of the first one and feed that back into `/stations/XXXX/observations/latest` to get the latest observations from the closest station to your point. In your case it is KMBL, so use `/stations/KMBL/observations/latest`.

**Reply by JJKraw** [reply]
Yes, this is the easiest way! I had forgotten the stations were returned with /points.

**Reply by jkreski** [reply]
OK, thanks! - I am able to trace it to:  https://api.weather.gov/gridpoints/APX/14,39/stations but when I store that URL to a variable within my FETCH construct and try to use that variable to FETCH it in another downstream FETCH construct it says it's not defined.  A scope problem.  I declare the variable globally but do it within the first FETCH construct.  Any Javascript ideas about how to work through this?  I am thinking maybe I have to somehow write it to a temporary file and read it back in, or maybe I am missing something.

**Reply by dwhitemv25** [reply]
General JavaScript programming questions probably belong more on sites like stackoverflow than here. But you shouldn't have to use a temp file, the global namespace in JS is global namespace, unless you're using some environment or library that changes the assumptions.

**Comment by JJKraw** [comment]
Look further at the data returned from the /point link in Michigan. It returns observationStations which is a link - request that URL and you get a list of all the observation stations for that office's coverage. Each station's data includes it's latitude and longitude so it's possible to figure out the closest station to any particular point.
